Frameworks/CoreText/CTLine.mm

```diff
@@ -16,16 +16,38 @@
 
 #import <CoreText/CTLine.h>
 #import <StubReturn.h>
-#import "NSStringInternal.h"
 #import "CoreTextInternal.h"
 #import "CGContextInternal.h"
 #import "DWriteWrapper_CoreText.h"
 #import <CoreText/CTTypesetter.h>
 
 #import <algorithm>
 #import <numeric>
+#import <wchar.h>
 #import <vector>
 
+static CFStringRef _CTCreateReversedString(CFStringRef string) {
+    if (string == nullptr) {
+        return nullptr;
+    }
+
+    CFIndex length = CFStringGetLength(string);
+    CFIndex usedBufLen;
+    CFStringGetBytes(string, CFRangeMake(0, length), kCFStringEncodingUTF16, 0, false, nullptr, length, &usedBufLen);
+
+    if (length < 2) {
+        return CFStringCreateCopy(kCFAllocatorDefault, string);
+    }
+
+    std::vector<UniChar> characters((usedBufLen / (sizeof(UniChar))) + 1, '\0');
```

Comment:
If you use unique_iw instead of std::vector, you can use `CFStringCreateWithBytesNoCopy` and detach it instead of copying the buffer.